Automattic/wp-calypso
# Save site migration credentials for auto migration

Related to https://github.com/Automattic/wp-calypso/issues/93532

## Proposed Changes

- Saves site credentials by calling the automated-migration cred saver API
- Adds a section to show general errors besides the field-specific errors
- Shows backend validation messages for both field-specific and general errors

## Why are these changes being made?


* To save the site creds

## Testing Instructions


- Apply this branch to your local environment or add use the Calypso Live link below.
- Open the dev console and activate the feature flag by running the following command:
`window.sessionStorage.setItem('flags', 'automated-migration/collect-credentials')`
- Apply the patch in your sandbox D158344-code
- Edit your `hosts` file to route your public-api.wordpress.com calls to your sandbox
- Now get into the flow using the `/start` URL 
- Go through the domain selection step
- Select the free plan on the Plans page
- On the Goals step, select the "Import existing content or website" option and click continue
- Input the source site URL in the Identify step
- Next, select the "Migrate Site" option on the Import or Migrate step
- Select "Do it for me" on the How to migrate step
- Go through the checkout
- Once you complete the checkout, you should land on the new Credentials step
- Try with different wrong inputs and make sure the validation works as expected
- Try with correct input following the attached video, make sure the creds get saved

https://github.com/user-attachments/assets/7c64ccd1-7242-46bf-933b-422f4f203f1d

## Pre-merge Checklist
